apps/docs.blocksense.network/components/common/Drawer.tsx

```diff
@@ -0,0 +1,195 @@
+
+interface DrawerContextValue {
+  isOpen: boolean;
+  toggleOpen: () => void;
+}
+
+const DrawerContext = createContext<DrawerContextValue>(
+  {} as DrawerContextValue,
+);
+
+export const Drawer = ({
+  children,
+  open,
+  onOpenChange,
+}: {
+  children: ReactNode;
+  open?: boolean;
+  onOpenChange?: (open: boolean) => void;
+}) => {
+  const [isOpen, setIsOpen] = useState<boolean>(open ?? false);
+
+  useEffect(() => {
+    if (open !== undefined) {
+      setIsOpen(open);
+    }
+  }, [open]);
+
+  const handleClickOutside = useCallback((event: Event) => {
+    const drawerElement = document.querySelector('.drawer__content');
+    if (drawerElement && !drawerElement.contains(event.target as Node)) {
+      handleClose();
+    }
+  }, []);
+
+  const handleKeyDown = useCallback((event: KeyboardEvent) => {
+    if (event.key === 'Escape') {
+      handleClose();
+    }
+  }, []);
+
+  useEffect(() => {
+    if (!isOpen) return;
+    document.addEventListener('mousedown', handleClickOutside);
+    document.addEventListener('keydown', handleKeyDown);
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+      document.removeEventListener('keydown', handleKeyDown);
+    };
+  }, [isOpen, handleClickOutside, handleKeyDown]);
+
+  const handleClose = () => {
+    setIsOpen(false);
+    onOpenChange?.(false);
+  };
+
+  const handleOpen = () => {
+    setIsOpen(true);
+    onOpenChange?.(true);
+  };
+
+  return (
+    <DrawerContext.Provider value={{ isOpen, toggleOpen: handleOpen }}>
+      {children}
+    </DrawerContext.Provider>
+  );
+};
+
+export const DrawerTrigger = ({
+  children,
+  asChild,
+}: {
+  children: ReactNode;
+  asChild?: boolean;
+}) => {
+  const { toggleOpen } = useContext(DrawerContext) ?? {};
+
+  if (!toggleOpen) {
+    throw new Error('DrawerTrigger must be used within a Drawer.');
+  }
+
+  if (asChild && children) {
+    return (children as ReactElement).props
+      ? React.cloneElement(children as ReactElement, {
+          onClick: (event: MouseEvent) => {
+            toggleOpen();
+            (children as ReactElement).props.onClick?.(event);
+          },
+        })
+      : null;
+  }
+
+  return (
+    <div onClick={toggleOpen} className="drawer__trigger cursor-pointer">
+      {children}
+    </div>
+  );
+};
+
+export const DrawerContent = ({
+  className,
+  children,
+}: {
+  className?: string;
+  children: ReactNode;
+}) => {
+  const { isOpen } = useContext(DrawerContext) ?? {};
+
+  if (!isOpen) return null;
+
+  return (
+    <div
+      className={cn(
+        'drawer fixed inset-0 z-50 flex items-end justify-center w-full',
+        className,
+      )}
+    >
+      <div className="drawer__background fixed inset-0 bg-black opacity-75" />
+      <section className="drawer__content inset-x-1.5 bottom-0 z-50 flex h-[52vh] p-2 flex-col rounded-t-[4px] border border-neutral-200 bg-zinc-50 dark:text-neutral-200 dark:bg-neutral-900 dark:border-neutral-600">
```

Comment:
```suggestion
      <section className="drawer__content w-full inset-x-1.5 bottom-0 z-50 flex h-[52vh] p-2 flex-col rounded-t-[4px] border border-neutral-200 bg-zinc-50 dark:text-neutral-200 dark:bg-neutral-900 dark:border-neutral-600">
```

This will resolve the width issue.

before
![image](https://github.com/user-attachments/assets/a1b64234-5aba-4a62-aa0d-e055d0644792)

after
![image](https://github.com/user-attachments/assets/87e5beb3-4579-48bf-a019-e34e5014ed2d)